File: frontend/types/stone.ts
export type StoneType = "strength" | "wisdom" | "charm" | "wealth" | "joy";

export type StoneRarity = "normal" | "shiny" | "rare" | "legendary";

export type StoneShape = "crystal";

export interface StoneTypeConfig {
  label: string;
  color: string;
  emoji: string;
  hp: number;
}

export const STONE_TYPES: Record<StoneType, StoneTypeConfig> = {
  strength: { label: "力量", color: "red", emoji: "🔴", hp: 5 },
  wisdom: { label: "智慧", color: "blue", emoji: "🔵", hp: 5 },
  charm: { label: "魅力", color: "pink", emoji: "🩷", hp: 5 },
  wealth: { label: "财富", color: "gold", emoji: "🟡", hp: 5 },
  joy: { label: "小确幸", color: "rainbow", emoji: "🌈", hp: 8 },
};

export interface Stone {
  id: string;
  userId: string;
  type: StoneType;
  color: string;
  content: string;
  imageUrl: string | null;
  aiTitle: string | null;
  aiMessage: string | null;
  rarity: StoneRarity;
  shape: StoneShape;
  isFavorite: boolean;
  drawCount: number;
  lastDrawnAt: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateStoneRequest {
  type: StoneType;
  content: string;
  imageUrl?: string;
}

export interface StoneListResponse {
  list: Stone[];
  total: number;
}

File: backend/app/schemas/stone.py
from datetime import datetime

from pydantic import BaseModel, Field


class CreateStoneRequest(BaseModel):
    type: str = Field(..., pattern=r"^(strength|wisdom|charm|wealth|joy)$")
    content: str = Field(..., min_length=1, max_length=2000)
    image_url: str | None = None


class StoneResponse(BaseModel):
    id: str
    type: str
    color: str
    content: str
    image_url: str | None
    ai_title: str | None
    ai_message: str | None
    rarity: str
    shape: str
    is_favorite: bool
    created_at: datetime

    class Config:
        from_attributes = True


class CreateStoneResponse(BaseModel):
    stone: StoneResponse
    hp_change: int
    hp_after: int


class StoneListResponse(BaseModel):
    list: list[StoneResponse]
    total: int
